src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,166 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> element = new Node<>(null, value, null);
```

Comment:
When creating a new `Node<T>`, pass all the fields to the `Node` constructor. Currently, you pass `null` for the previous and next nodes, which should be the current tail and `null` respectively.